src/inference_fixes.jl

```diff
@@ -1,24 +1,25 @@
 # In this file we define special-cases to prevent Cassette related inference issues
 
-const BLACK_LIST = (
-    Base.promote_op, Base.to_shape,
+for F in [
+    Base.promote_op, 
+    Base.to_shape,
     Core.getfield,
     Core.:(===),
     Base.iterate,
     Broadcast.broadcasted,
     Broadcast.preprocess,
     Broadcast.combine_axes,
+    Base.copyto!,
```

Comment:
`copyto!` here is a blunt tool no doubt but helps make the CUDA code easier, plus maybe its a fair tradeoff to assume that once we're in the mutating version of this no new arrays are getting creating and so we can stop overdubbing for performance?

Reply:
I think that is reasonable